src/domain/relay/contracts/safe-contract.helper.spec.ts

```diff
@@ -0,0 +1,154 @@
+    },
+    {
+      internalType: 'uint256',
+      name: 'gasPrice',
+      type: 'uint256',
+    },
+    {
+      internalType: 'address',
+      name: 'gasToken',
+      type: 'address',
+    },
+    {
+      internalType: 'address payable',
+      name: 'refundReceiver',
+      type: 'address',
+    },
+    {
+      internalType: 'bytes',
+      name: 'signatures',
+      type: 'bytes',
+    },
+  ],
+  name: 'execTransaction',
+  outputs: [
+    {
+      internalType: 'bool',
+      name: 'success',
+      type: 'bool',
+    },
+  ],
+  stateMutability: 'payable',
+  type: 'function',
+};
+
+describe('Safe Contract Helper Tests', () => {
+  let target: SafeContractHelper;
+
+  beforeEach(() => {
+    target = new SafeContractHelper();
+  });
+
+  it('decodes an execTransaction correctly', () => {
+    const to = getAddress(faker.finance.ethereumAddress());
+    const value = faker.number.bigInt();
+    const callData = getExecTransactionCallData({ to, value });
+
+    const actual = target.decode(SafeContractHelper.EXEC_TRANSACTION, callData);
+
+    expect(actual).toEqual({ data: '0x00', to: to, value: value });
+  });
+
+  it('decoding a non execTransaction call throws', () => {
+    const functionSignature = faker.string.hexadecimal({
+      length: 8,
+    }) as `0x${string}`;
+    const arg1 = pad(getAddress(faker.finance.ethereumAddress()));
+    const arg2 = pad(getAddress(faker.finance.ethereumAddress()));
+    const callData = concatHex([functionSignature, arg1, arg2]);
+
+    expect(() => {
+      target.decode(SafeContractHelper.EXEC_TRANSACTION, callData);
+    }).toThrow();
+  });
+
+  it('isCall returns true for a execTransaction call', () => {
+    const callData = getExecTransactionCallData();
+
+    const actual = target.isCall(callData);
+
+    expect(actual).toBe(true);
+  });
+
+  it('isCall returns false for a non safe call', () => {
+    const functionSignature = faker.string.hexadecimal({
+      length: 8,
+    }) as `0x${string}`;
+    const arg1 = pad(getAddress(faker.finance.ethereumAddress()));
+    const arg2 = pad(getAddress(faker.finance.ethereumAddress()));
+    const callData = concatHex([functionSignature, arg1, arg2]);
+
+    const actual = target.isCall(callData);
+
+    expect(actual).toBe(false);
+  });
+});
+
+function getExecTransactionCallData(args?: {
+  to?: `0x${string}`;
+  value?: bigint;
+}): `0x${string}` {
+  const to = args?.to ?? getAddress(faker.finance.ethereumAddress());
+  const value = args?.value ?? faker.number.bigInt();
+  const data = toHex(0);
+  const operation = CALL_OPERATION;
+  const safeTxGas = faker.number.bigInt();
+  const baseGas = faker.number.bigInt();
+  const gasPrice = faker.number.bigInt();
+  const gasToken = getAddress(faker.finance.ethereumAddress());
+  const refundReceiver = getAddress(faker.finance.ethereumAddress());
+  const signatures = faker.string.hexadecimal({ length: 64 * 2 });
+
+  return encodeFunctionData({
+    abi: [ABI_ITEM],
```

Comment:
Instead of manually adding the ABI here, we could call the deployment getter inside this function.

Reply:
Done in [bdcbda1](https://github.com/safe-global/safe-client-gateway/pull/680/commits/bdcbda1c2d7e8987b4bd3d5e4ccd3cfdb28610dd)